### PR TITLE
[matter_yamltests] Add supports for OTA_SuccessfulTransfer specific S…

### PR DESCRIPTION
**Pull request by vivien-apple**
…ystemCommands and DelayCommands

#### Problem

`OTA_SuccessfulTransfer` has custom `SystemCommands/DelayCommands` it needs to runs the tests. This PR adds them and basically copy the content of `src/app/tests/suites/commands/*/scripts/[WaitForMessage|CompareFiles].py`

### CODE REVIEW

**Review by jtung-apple — APPROVED** [review]

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py [hunk 1]
```diff
@@ -13,64 +13,29 @@
...
 import sys
-import xmlrpc.client
 
 from ..pseudo_cluster import PseudoCluster
+from .accessory_server_bridge import AccessoryServerBridge
 
-DEFAULT_KEY = 'default'
-IP = '127.0.0.1'
-PORT = 9000
+DEFAULT_CHIP_ROOT = os.path.abspath(
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: I told you that it should be a package ;) 
More seriously, that is annoying. I think this is OK for now as we need to move forward but we need to find a proper solution at the end of the day.

[the remark this answers is not in the document]
Comment (tehampson) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: Even if this was it's own package of yamlclusters you would still need to solve this are you are grabbing a file that is outside of the packages using a relative path that is outside of `<chip_root>/scripts`

If we are time bound we could let this in as is, but it will bite us later on when TH team is trying to port this using package. But, I do think that this can be fixed with 1 hour of work to properly allow for code reuse and python code package distribution either:
1. Package `/src/app/ota_image_tool.py` to be it's own package which yamltests will depend on.
2. Take out the required code that both sides need and place that in its own package inside scripts, and have both that script and yamltests depend on this new package.

[the remark this answers is not in the document]
Comment (tehampson) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: Wait I just realized you are using relative path based on where this happens to get installed in the virtual enviroment. This is not even relative to where this file sits in the repo here. I think we need to fix this before merging

[the remark this answers is not in the document]
Comment (andy31415) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: if we run under `bootstrap/activate` then `$PW_PROJECT_ROOT` is set. Or otherwise lets make some env variable or command line argument override available.

[the remark this answers is not in the document]
Comment (andy31415) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: I would suggest some `_GetChipRoot` that checks various env variables (maybe a dedicated one, followed by a PW_PROJECT_ROOT fallback and if that fails, relative path). I would also check for some known file existence and print out a friendly exception if the check fails.

Allow the user to bypass things.

[the remark this answers is not in the document]
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py:24]: Sorry for the lag on this. I will come up with something in the next few days.

[the remark this answers is not in the document]
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/system_commands.py]: Ok. I moved the related OTA specific function code under `scripts/tests/chiptest` for 2 reasons.
The first one is to avoid duplicating the specific `CompareFiles.py` and `CreateOTAImage.py` code.
The second is because in my case the test harness (in this case `scripts/tests/chiptest`) can be remote, so it needs to run on the remote.

File: src/app/tests/suites/commands/system/scripts/CreateOtaImage.py [hunk 2]
```diff
@@ -14,32 +14,20 @@
...
-    otaImageTool = DEFAULT_CHIP_ROOT + '/src/app/ota_image_tool.py'
-    cmd = [otaImageTool, 'create', '-v', '0xDEAD', '-p', '0xBEEF', '-vn', '2',
-           '-vs', "2.0", '-da', 'sha256', rawImageFilePath, otaImageFilePath]
-    s = subprocess.Popen(cmd)
-    s.wait()
-    if s.returncode != 0:
-        raise Exception('Cannot create OTA image file')
-
-
-if __name__ == "__main__":
```
Comment (tehampson) [inline src/app/tests/suites/commands/system/scripts/CreateOtaImage.py:44]: Why did you remove this (`if __name__ == "__main__":`...)? I am assuming that you can just keep this as my understanding this is the pythonic way. Same with the one in the file `src/app/tests/suites/commands/system/scripts/CompareFiles.py`

**Review by tehampson — COMMENTED** [review]
Overall LGTM, I just want to think about this just a little longer, but added one nit comment for the time being

File: src/app/tests/suites/commands/system/scripts/CreateOtaImage.py (hunk 2 above)
Comment (vivien-apple) [reply src/app/tests/suites/commands/system/scripts/CreateOtaImage.py:44]: Well, there is no `main` anymore. Those scripts looks like the others that are sending command over xml-rpc. It may be the pythonic way to do it and I'm not arguing otherwise but it would be better to do an other PR to fix all those scripts if we really want. That said, that looks like lost time since this is not used often and we may remove those once we are done ;)

Comment (tehampson) [reply src/app/tests/suites/commands/system/scripts/CreateOtaImage.py:44]: Fair enough... Precedent of the other scripts doesn't have a main

**Review by tehampson — APPROVED** [review]
